static/app/components/dropdownMenuControlV2.tsx

```diff
@@ -0,0 +1,193 @@
+import {useMenuTriggerState} from '@react-stately/menu';
+import {MenuTriggerProps} from '@react-types/menu';
+
+import DropdownButton, {DropdownButtonProps} from 'sentry/components/dropdownButtonV2';
+import {MenuItemProps} from 'sentry/components/dropdownMenuItemV2';
+import Menu from 'sentry/components/dropdownMenuV2';
+
+type TriggerProps = {
+  props: HTMLAttributes<HTMLButtonElement> & {
+    onClick?: (e: MouseEvent) => void;
+  };
+  ref: RefObject<HTMLButtonElement>;
+};
+
+type Props = {
+  /**
+   * Items to display inside the dropdown menu. If the item has
+   * a `children` prop, it will be rendered as a menu section.
+   * If it has a `children` prop and its `isSubmenu` prop
+   * is true, it will be rendered as a submenu.
+   */
+  items: MenuItemProps[];
+  /**
+   * By default, the menu trigger will be rendered as a button,
+   * with triggerLabel as the button label.
+   */
+  triggerLabel?: ReactNode;
+  /**
+   * If using the default button trigger (i.e. the custom `trigger` prop
+   * has not been provided), then `triggerProps` will be passed on
+   * to the button component.
+   */
+  triggerProps?: DropdownButtonProps;
+  /**
+   * Optionally replace the trigger button with a different component.
+   * Note that the replacement must have the `props` and `ref` (supplied
+   * in TriggerProps) forwarded its outer wrap, otherwise the accessibility
+   * features won't work correctly.
+   */
+  trigger?: (props: TriggerProps) => ReactNode;
+  /**
+   * Whether the trigger is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Title to put on top of
+   */
+  menuTitle?: string;
+  /**
+   * Whether this is a submenu.
+   */
+  isSubmenu?: boolean;
+  /**
+   * If this is a submenu, it will in some cases need to close the root menu
+   * (e.g. when a submenu item is clicked).
+   */
+  closeRootMenu?: () => void;
+  /**
+   * If this is a submenu, it will in some cases need to close itself
+   * (e.g. when the user presses the arrow left key)
+   */
+  closeCurrentSubmenu?: () => void;
+  /**
+   * Tag name for the outer wrap, defaults to `div`
+   */
+  renderWrapAs?: ElementType;
+  /**
+   * Pass class name to the outer wrap
+   */
+  className?: string;
+} & Partial<MenuTriggerProps> &
+  Partial<AriaMenuOptions<MenuItemProps>> &
+  Partial<OverlayProps> &
+  Partial<AriaPositionProps>;
+
+/**
+ * A menu component that renders both the trigger button
+ * and the dropdown menu.
+ * See: https://react-spectrum.adobe.com/react-aria/useMenuTrigger.html
+ */
+function MenuControl({
+  items,
+  trigger,
+  triggerLabel,
+  triggerProps = {},
+  isDisabled,
+  isSubmenu = false,
+  closeRootMenu,
+  closeCurrentSubmenu,
+  renderWrapAs = 'div',
+  className,
+  ...props
+}: Props) {
+  const ref = useRef(null);
+
+  /**
+   * Control the menu open state
+   * See: https://react-spectrum.adobe.com/react-aria/useMenuTrigger.html
+   */
+  const state = useMenuTriggerState(props);
```

Comment:
nit, but my general usage recommendation of the "flag" style comments is for documenting types, constants, basically anything that lives at the top level of a module (.ts file). 

So in this case I would write this as 

```suggestion

  // Control the menu open state
  // See: https://react-spectrum.adobe.com/react-aria/useMenuTrigger.html
  const state = useMenuTriggerState(props);
```